Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -0,0 +1,135 @@
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            if (!link->first_node("inertial"))
+            {
+                links.push_back(link);
+            }
+        }
+
+        for (auto* link : links)
+        {
+            xml_node<>* inertial = urdf->document()->allocate_node(node_element, "inertial");
+
+            xml_node<>* mass = urdf->document()->allocate_node(node_element, "mass");
+            mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
+            inertial->append_node(mass);
+            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
+
+            inertial->append_node(inertia);
+
+            xml_node<>* origin = urdf->document()->allocate_node(node_element, "origin");
+            origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
+            inertial->append_node(origin);
+
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                modifiedLinks.push_back(name->value());
+            }
+            link->append_node(inertial);
+        }
+
+        return modifiedLinks;
+    }
+
+    AZStd::vector<AZStd::string> ChangeDuplications(AZ::rapidxml::xml_node<>* urdf)
+    {
+        using namespace AZ::rapidxml;
+        AZStd::vector<AZStd::string> modifiedLinks;
+        AZStd::set<AZStd::string> linkAndJointsName;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                linkAndJointsName.insert(name->value());
+            }
+        }
+        for (xml_node<>* joint = urdf->first_node("joint"); joint; joint = joint->next_sibling("joint"))
+        {
+            auto* name = joint->first_attribute("name");
+            if (name)
+            {
+                if (linkAndJointsName.contains(name->value()))
+                {
+                    auto newName = AZStd::string(name->value()) + "_joint_dup";
```

Comment:
I am wondering whether it is priority to keep joint names unchanged (as opposed to keeping link names unchanged) since we actually use joint names for joint manipulation. However, we create entity names and derive namespaces based on links. What do you think?

Reply:
Both are important. Links names affect frame names, other joint names.
Joint names are easier to change IHMO in frame component.